libraries/plugins/xfpga/sysfs.c

```diff
@@ -1807,8 +1828,35 @@ fpga_result get_fpga_hw_type(fpga_handle handle, enum fpga_hw_type *hw_type)
 		goto out_unlock;
 	}
 
+	if (snprintf(sysfs_path, SYSFS_PATH_MAX,
```

Comment:
These values are also in the [token header](https://github.com/OFS/opae-sdk/blob/master/include/opae/types.h#L276-L277).  You could save some time by retrieving them from the token.